test/e2e/operator/deployment_api.go

```diff
@@ -223,6 +227,175 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 				return validate.DriverDeployment(client, k8sver, d.Namespace, deployment)
 			}, "1m", "20s").ShouldNot(HaveOccurred(), "driver validation failure after restarting")
 		})
+
+		It("shall allow to revert back to previous driver mode", func() {
+			driverName := "switch-driver-mode"
+			deployment := api.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: driverName,
+				},
+				Spec: api.DeploymentSpec{
+					DeviceMode:     api.DeviceModeLVM,
+					PMEMPercentage: 50,
+				},
+			}
+
+			deployment = deploy.CreateDeploymentCR(f, deployment)
+			defer deploy.DeleteDeploymentCR(f, deployment.Name)
+			validateDriver(deployment)
+
+			// NOTE(avalluri): As the current operator does not support deploying
+			// the driver in 'testing' mode, we cannot directely access CSI
+			// interface of it. Hence, using SC/PVC for creating volumes.
+			//
+			// Once we add "-testing" support we could simplify the code
+			// by using controller's CSI interface to create/delete/publish
+			// the test volume.
+
+			reclaim := corev1.PersistentVolumeReclaimDelete
+			immediate := storagev1.VolumeBindingImmediate
+
+			// Create storage class
+			sc := &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "switch-mode-sc",
+				},
+				Provisioner:       driverName,
+				ReclaimPolicy:     &reclaim,
+				VolumeBindingMode: &immediate,
+			}
+
+			Eventually(func() error {
+				_, err := f.ClientSet.StorageV1().StorageClasses().Create(context.Background(), sc, metav1.CreateOptions{})
+				deploy.LogError(err, "create storage class error: %v, will retry...", err)
+				return err
+			}, "3m", "1s").Should(BeNil(), "create storage class %q", sc.Name)
```

Comment:
For errors it is better to use `ShouldNot(HaveOccurred())` instead of `BeNil`. It's more specific and I think the output is better when it fails.

This also applies to the rest of the function.